src/videoCache.js

```diff
@@ -0,0 +1,213 @@
+}
+
+/**
+ * Wraps a bid in the format expected by the prebid-server endpoints, or returns null if
+ * the bid can't be converted cleanly.
+ *
+ * @param {CacheableBid} bid
+ */
+function toStorageRequest(bid) {
+  return {
+    'value': wrapURI(bid.vastUrl)
+  };
+}
+
+/**
+ * If the server responded successfully, then it gave us a bunch of objects with uuid properties.
+ * Abstract our background dependency away by renaming them to cacheId.
+ */
+function fromStorageResponse(response) {
+  return {
+    cacheId: response.uuid
+  };
+}
+
+/**
+ * A function which should be called with the results of the storage operation.
+ *
+ * @callback videoCacheStoreCallback
+ *
+ * @param {Error} [error] The error, if one occurred.
+ * @param {?string[]} uuids An array of unique IDs. The array will have one element for each bid we were asked
+ *   to store. It may include null elements if some of the bids were malformed, or an error occurred.
+ *   Each non-null element in this array is a valid input into the retrieve function, which will fetch
+ *   some VAST XML which can be used to render this bid's ad.
+ */
+
+/**
+ * A function which bridges the APIs between the videoCacheStoreCallback and our ajax function's API.
+ *
+ * @param {videoCacheStoreCallback} callback A callback to the "store" function.
+ * @return {Function} A callback which interprets the cache server's responses, and makes up the right
+ *   arguments for our callback.
+ */
+function shimStorageCallback(callback) {
+  return {
+    success: function(responseBody) {
+      let ids;
+      try {
+        ids = JSON.parse(responseBody).responses.map(fromStorageResponse)
+      }
+      catch (e) {
+        callback(e, []);
+        return;
+      }
+
+      callback(null, ids);
+    },
+    error: function(statusText, responseBody) {
+      callback(new Error('Error storing video ad in the cache: ' + statusText + ': ' + JSON.stringify(responseBody)), []);
+    }
+  }
+}
+
+/**
+ * If the given bid is for a Video ad, generate a unique ID and cache it somewhere server-side.
+ *
+ * @param {CacheableBid[]} bids A list of bid objects which should be cached.
+ * @param {videoCacheStoreCallback} [callback] An optional callback which should be executed after
+ *   the data has been stored in the cache.
+ */
+export function store(bids, callback) {
+  const requestData = {
+    puts: bids.map(toStorageRequest)
+  };
+
+  ajax(PUT_URL, shimStorageCallback(callback), JSON.stringify(requestData), {
+    contentType: 'text/plain',
+    withCredentials: true
+  });
+}
+
+/**
+ * @callback videoCacheRetrieveCallback
+ *
+ * @param {Error} [error] The error, if one occurred.
+ * @param {string} A VAST XML string. If error is defined, this can still expect an "empty"
+ *   VAST URL response.
+ */
+
+/**
+ * Takes the user-defined callback, and wraps it into our ajax function's API.
+ *
+ * This also needs to work around the fact that the cache server returns a 200 on "data not found"...
```

Comment:
er... and one other judgment call which needs agreement:

1. If the videoCache can't find the video, it should be an error.

(I didn't see any discussion on this point... but it's an important one. If you agree, then a prebid-cache 204 "success" would still be considered an "error" from this file's point of view, and all this code would still be necessary anyway)